System/Process.hsc

```diff
@@ -413,13 +414,27 @@ readProcess
     -> [String]                 -- ^ any arguments
     -> String                   -- ^ standard input
     -> IO String                -- ^ stdout
-readProcess cmd args input = do
-    let cp_opts = (proc cmd args) {
+readProcess cmd args = readCreateProcess $ proc cmd args
+
+-- | @readCreateProcess@ works exactly like 'readProcess' except that it
+-- lets you pass 'CreateProcess' giving better flexibility.
+--
+-- >  > readCreateProcess (shell "pwd" { cwd = "/etc/" }) ""
+-- >  "/etc\n"
+--
+-- /Since: 1.2.3.0/
+
+readCreateProcess
+    :: CreateProcess
```

Comment:
I think it's worth commenting that any values for `std_in`, `std_out` and `std_err` will be overwritten. Additionally: is it worth including `readCreateProcessWithExitCode` as well?